### PR TITLE
added SyncIQ replication job

### PR DESCRIPTION
**Pull request by shenda1**
# Description
PR is raised to add SyncIQ replication job data source.

# ISSUE TYPE

- Feature Pull Request

##### RESOURCE OR DATASOURCE NAME
powerscale_synciq_replication_job

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Unit tests
- [x] Acceptance tests

image

### CODE REVIEW

**Comment by shenda1** [comment]
run e2e test SyncIQReplicationJobDataSource

File: powerscale/helper/synciq_replication_job_helper.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+		}
+	}
+
+	resp, _, err := jobParams.Execute()
+	if err != nil {
+		return nil, err
+	}
+
+	// Pagination
+	if resp.Resume != nil {
```
Comment (rounak-adhikary) [inline powerscale/helper/synciq_replication_job_helper.go:59]: You also need to check that user has NOT provided `limit`.
Reference logic: https://github.com/dell/terraform-provider-powerscale/commit/e100f6c3be6eb49b1242a5817b8e173f1da218dc#diff-09404796aadc28199b3968e46808a855ecf08170250d9ac48433bc83646af9f8R52

File: powerscale/helper/synciq_replication_job_helper.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+	var jobs []models.SyncIQReplicationJobModel
+	for _, job := range response.Jobs {
+		var temp models.SyncIQReplicationJobModel
+
+		err = CopyFieldsToNonNestedModel(ctx, job, &temp)
+		if err != nil {
+			resp.AddError(
+				"Unable to update SyncIQ replication job state",
+				fmt.Sprintf("Unable to update SyncIQ replication job state with error %s", err),
```
Comment (rounak-adhikary) [inline powerscale/helper/synciq_replication_job_helper.go:92]:
```suggestion
				fmt.Sprintf("Unable to update SyncIQ replication job state with error %s", err.Error()),
```

**Comment by shenda1** [comment]
run e2e test SyncIQReplicationJobDataSource

**Review by rounak-adhikary — APPROVED** [review]

**Review by Krunal-Thakkar — APPROVED** [review]

**Pull request merged**